source/SonicHeroes.WidescreenFix/dllmain.cpp

```diff
@@ -121,8 +116,45 @@ void Init()
                 fmul    dword ptr[eax08]
             }
         }
-    }; injector::MakeInline<CutOffAreaHook>(pattern.count(1).get(0).get<uint32_t>(0), pattern.count(1).get(0).get<uint32_t>(17));
-
+    }; injector::MakeInline<CutOffAreaHORHook>(pattern.count(1).get(0).get<uint32_t>(0), pattern.count(1).get(0).get<uint32_t>(17));
+    
+    struct CutOffAreaVertHook
+	{
+		void operator()(injector::reg_pack& regs)
+		{
+            float edx6C = *(float*)(regs.edx + 0x6C) / Screen.fHudZoomScale;
+            float eax10 = *(float*)(regs.eax + 0x10);
+            _asm
+            {
+                fld     dword ptr[edx6C]
+                fmul    dword ptr[eax10]
+            }
+         }
+    }; injector::MakeInline<CutOffAreaVertHook>(pattern.count(1).get(0).get<uint32_t>(42), pattern.count(1).get(0).get<uint32_t>(6));
```

Comment:
You can just write 'pattern.get_first(42)', much shorter that way.

Reply:
Yeah but the second one's for how much to overwrite with nops and the first is the offset.

Reply:
Um, it's incorrect. First, you specify an address where you will place the hook, second one is the address to where the hook jumps back. Everything in between is filled with nops.

Reply:
Oh ok. Do you have the header files to windows.h, SDKDDKVer.h and subauth? I need those so I can compile it in Visual Studio in-order to try it to see if it works.

Reply:
What do you mean? You can compile everything as is, just like appveyor can. If it throws errors, try changing sdk version in project's settings.